### PR TITLE
Setup and Cleanup for softwareraid test

### PR DESCRIPTION
**Pull request by narasimhan-v**
Considering a requirement of just creating / deleting a raid.
This is useful if some tests needs to be run on created raid.

Signed-off-by: Narasimhan V <sim@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by narasimhan-v** [comment]
```
# avocado run softwareraid.py -m softwareraid.py.data/softwareraid.yaml;  cat /proc/mdstat; avocado run softwareraid.py -m softwareraid.py.data/softwareraid_setup.yaml; cat /proc/mdstat; avocado run softwareraid.py -m softwareraid.py.data/softwareraid_cleanup.yaml; cat /proc/mdstat; avocado run softwareraid.py -m softwareraid.py.data/softwareraid.yaml
JOB ID     : 1746b05bf82bdb84b46a976d8629b632d297922a
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2018-03-11T04.33-1746b05/job.log
 (1/3) softwareraid.py:SoftwareRaid.test_run;raidlinear-scenario-675a: PASS (1.03 s)
 (2/3) softwareraid.py:SoftwareRaid.test_run;raid0-scenario-1d7e: PASS (1.02 s)
 (3/3) softwareraid.py:SoftwareRaid.test_run;raid1-scenario-07be: PASS (152.17 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 155.00 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2018-03-11T04.33-1746b05/results.html
Personalities : [raid1] [raid0] [linear] 
unused devices: <none>


JOB ID     : d7588ddf0856665fee0aea5ae28795e2926454b1
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2018-03-11T04.36-d7588dd/job.log
 (1/1) softwareraid.py:SoftwareRaid.test_run;raid1-4ac9: PASS (0.73 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 1.16 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2018-03-11T04.36-d7588dd/results.html
Personalities : [raid1] [raid0] [linear] 
md125 : active raid1 nvme0n32[2](S) nvme0n31[1] nvme0n29[0]
      29982592 blocks super 1.2 [2/2] [UU]
unused devices: <none>


JOB ID     : f73f94f5fdc131d81b0f65f77c5516171c93c740
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2018-03-11T04.36-f73f94f/job.log
 (1/1) softwareraid.py:SoftwareRaid.test_run;raid1-192a: PASS (0.63 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 1.00 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2018-03-11T04.36-f73f94f/results.html
Personalities : [raid1] [raid0] [linear] 

unused devices: <none>


JOB ID     : 84a23442511353cf1e3360ef8f6c86e7e8ca8e53
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2018-03-11T04.36-84a2344/job.log
 (1/3) softwareraid.py:SoftwareRaid.test_run;raidlinear-scenario-675a: PASS (1.04 s)
 (2/3) softwareraid.py:SoftwareRaid.test_run;raid0-scenario-1d7e: PASS (1.10 s)
 (3/3) softwareraid.py:SoftwareRaid.test_run;raid1-scenario-07be: PASS (152.27 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 155.11 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2018-03-11T04.36-84a2344/results.html
#
```

**Comment by PraveenPenguin** [comment]
@vrbagalkote can you please review this PR

File: io/disk/softwareraid.py [hunk 1]
```diff
@@ -66,19 +70,22 @@ def test_run(self):
...
         Only basic operations are run viz create and delete
         """
         cmd = "echo 'yes' | mdadm --create --verbose --assume-clean \
-            /dev/md/mdsraid --level=%s --raid-devices=%d %s \
-            --force" \
+            /dev/md/mdsraid --level=%s --raid-devices=%d %s" \
             % (self.raidlevel, self.disk_count, self.disk)
+        if self.sparedisk:
+            cmd += " --spare-devices=1 %s " % self.sparedisk
+        cmd += " --force"
```
Comment (vrbagalkote) [inline io/disk/softwareraid.py]: nice

File: io/disk/softwareraid.py.data/Readme [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
 * Name of the devices on which this scripts should be run.
 
 * Raid levels to be created
 
+* Setup - True if raid needs to be created
+
+* Cleanup - True if raid needs to be deleted
+
+* Run_tests - True if mdadm tests needs to be run on created raid
+
```
Comment (vrbagalkote) [inline io/disk/softwareraid.py.data/Readme]: confusing...U create raid only from mdadm...can u explain, what u mean

Comment (narasimhan-v) [reply io/disk/softwareraid.py.data/Readme]: If setup and cleanup is set to False, and run_tests is set to True, then only the mdadm tests are run on it.

Comment (vrbagalkote) [reply io/disk/softwareraid.py.data/Readme]: can u rephrase it properly

Comment (narasimhan-v) [reply io/disk/softwareraid.py.data/Readme]: I am not sure what part you did not understand here.
So, it is better if you suggest the text.

Comment (vrbagalkote) [reply io/disk/softwareraid.py.data/Readme]: ```Run_tests - True if mdadm tests needs to be run on created raid``` 
This confusing, you create raid using mdadm...and what mdadm tests you run on raid created with mdadm

Comment (narasimhan-v) [reply io/disk/softwareraid.py.data/Readme]: fail, remove, add device

**Comment by narasimhan-v** [comment]
@vrbagalkote Updated the Readme. Please review.

**Comment by vrbagalkote** [comment]
@narasimhan-v thanks...I think, its better now...can be merged..

**Comment by PraveenPenguin** [comment]
@narasimhan-v thanks for patch , @vrbagalkote thanks for ACK , merging now

**Pull request merged**